tests/transforms/augmentation/test_random_gamma.py

```diff
@@ -0,0 +1,31 @@
+from torchio import RandomGamma
+from ...utils import TorchioTestCase
+from numpy.testing import assert_array_equal
+
+
+class TestRandomGamma(TorchioTestCase):
+    """Tests for `RandomGamma`."""
+    def test_with_zero_gamma(self):
+        transform = RandomGamma(log_gamma_range=0)
+        transformed = transform(self.sample)
+        assert_array_equal(self.sample.t1.data, transformed.t1.data)
+
+    def test_with_non_zero_gamma(self):
+        transform = RandomGamma(log_gamma_range=(0.1, 0.3))
+        transformed = transform(self.sample)
+        with self.assertRaises(AssertionError):
+            assert_array_equal(self.sample.t1.data, transformed.t1.data)
+
+    def test_with_high_gamma(self):
+        transform = RandomGamma(log_gamma_range=(100, 100))
+        transformed = transform(self.sample)
+        assert_array_equal(self.sample.t1.data == 1, transformed.t1.data)
+
+    def test_with_low_gamma(self):
+        transform = RandomGamma(log_gamma_range=(-100, -100))
+        transformed = transform(self.sample)
+        assert_array_equal(self.sample.t1.data > 0, transformed.t1.data)
```

Comment:
Can you please explain these?

Reply:
It is related to the `..note` I will have to add but as we give a log range, the power at which we raise the intensity can be quickly very low or very high pushing towards extreme values. Therefore, if we set `log_gamma_range=(100, 100)` with intensity values between 0 and 1, all non zero values will be pushed to 1 when raised to the power gamma. It is just a test to make sure it behaves well regarding high or low values of gamma.

Reply:
Hmm let's see if I understand.

Sample is `[0.0 0.1, 0.5, 0.9, 1.0]`

**If log(gamma) = 100,**
then gamma = e^100 ~ 2.68^43 -> values in [0, 1) go to 0: transformed is `[0, 0, 0, 0, 1]`. That's why you compare sample == 1 with the result.

**If log(gamma) = -100,**
then gamma = e^-100 ~ 0 -> values in [0, 1) go to 1: transformed is `[1, 1, 1, 1, 1]`. Note that 0^0 = 1 in Python, so I think all values will go to 1.

1. I think you can also compare the rest of the values, making sure they are **almost** equal to the expected values.

Reply:
Yes that's it, OK, I will compare all values

Reply:
In fact I am already looking at all the values because I compare the boolean maps to the whole tensors. What were you thinking when you were talking about comparing the rest of the values?